test/test_client.py

```diff
@@ -0,0 +1,77 @@
+from collections import Counter
+from unittest import TestCase
+import logging
+
+from mock import Mock, patch
+
+from apns2.client import APNsClient, CONCURRENT_STREAMS_SAFETY_MAXIMUM
+from apns2.payload import Payload
+
+class ClientTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        logging.basicConfig()
+        cls.tokens = ["%064x" % i for i in range(10000)]
+        cls.notification = Payload(alert="Hello, world!")
+        cls.topic = "com.example.App"
+
+    
+    def setUp(self):
+        self.open_streams = 0
+        self.max_open_streams = 0
+        self.mock_results = None
+        
+        with patch("apns2.client.HTTP20Connection") as mock_connection_constructor, \
+            patch("apns2.client.init_context"):
+            self.mock_connection = Mock()
+            self.mock_connection.get_response.side_effect = self.mock_get_response
```

Comment:
Sorry, I guess what I meant was to use a good old class for mock_connection.  With mock_connection, you're defining methods and using instance variables; I think it might be much more straightforward with a good old class.  The Mock library probably makes sense for the other mocks.

Also, I hate the mock.side_effect attribute.  I think I'm almost used to reading it and using it by now, but I still think it's confusing and counterintuitive.